tests/client/executor_client.c

```diff
@@ -23,33 +20,39 @@ void start_executor (char *student_code)
 			printf("chdir: %s\n", strerror(errno));
 		}
 
-		//add the directory with all the student code to PYTHONPATH if it's not already there
-		if ((python_path = getenv("PYTHONPATH")) == NULL) { //if PYTHONPATH is not defined
-			new_python_path = malloc(strlen(python_path_header) + strlen(path_to_test_student_code) + 1);
+		// add the directory with all the student code to PYTHONPATH if it's not already there
+		if ((python_path = getenv(PYTHONPATH)) == NULL) { // if PYTHONPATH is not defined
+			new_python_path = malloc(strlen(PYTHONPATH) + 1 + strlen(path_to_test_student_code) + 1);
 			new_python_path[0] = '\0';
-			strcat(new_python_path, python_path_header);
+			strcat(new_python_path, PYTHONPATH);
+			strcat(new_python_path, "=");
 			strcat(new_python_path, path_to_test_student_code);
-		} else if (strstr(python_path, "tests/student_code") == NULL) { //if PYTHONPATH is defined but the path we want isn't there
-			new_python_path = malloc(strlen(python_path) + strlen(path_to_test_student_code) + strlen(python_path_header) + 1);
+		} else { // if PYTHONPATH is defined
+			new_python_path = malloc(strlen(PYTHONPATH) + 1 + strlen(python_path) + 1 + strlen(path_to_test_student_code) + 1);
```

Comment:
you actually don't need the intermediate +1s because `strcat` will concatenate the two strings together, with the first character of the second string starting at the location of the terminating null character of the first string. But this should still be fine because we're allocating more memory than needed